cdc/capture/http_handler.go

```diff
@@ -0,0 +1,620 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package capture
+
+import (
+	"bufio"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/pingcap/ticdc/pkg/config"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pingcap/ticdc/cdc/kv"
+	"github.com/pingcap/ticdc/cdc/model"
+	"github.com/pingcap/ticdc/cdc/owner"
+	cerror "github.com/pingcap/ticdc/pkg/errors"
+	chttputil "github.com/pingcap/ticdc/pkg/httputil"
+	"github.com/pingcap/ticdc/pkg/version"
+	"github.com/tikv/client-go/v2/oracle"
+)
+
+const (
+	// APIOpVarChangefeedState is the key of changefeed state in HTTP API
+	APIOpVarChangefeedState = "state"
+	// APIOpVarChangefeedID is the key of changefeed ID in HTTP API
+	APIOpVarChangefeedID = "changefeed_id"
+	// APIOpVarCaptureID is the key of capture ID in HTTP API
+	APIOpVarCaptureID = "capture_id"
+	// APIOpVarTableID is the key of table ID in HTTP API
+	APIOpVarTableID = "table_id"
+	// ForwardFromCapture is a header to be set when a request is forwarded from another capture
+	ForwardFromCapture = "TiCDC-ForwardFromCapture"
+)
+
+type httpHandler struct {
+	capture *Capture
+}
+
+// NewHTTPHandler return a httpHandler for OpenAPI
+func NewHTTPHandler(capture *Capture) httpHandler {
```

Comment:
The method is exported, but the returned structure is not exported. Can you confirm that you are keeping all methods and fields minimally visible?